packages/edit-site/src/components/editor/global-styles-renderer.js

```diff
@@ -13,39 +13,46 @@ import { __EXPERIMENTAL_STYLE_PROPERTY as STYLE_PROPERTY } from '@wordpress/bloc
  */
 import { PRESET_CATEGORIES, LINK_COLOR_DECLARATION } from './utils';
 
-const mergeTrees = ( baseData, userData ) => {
+export const mergeTrees = ( baseData, userData ) => {
 	// Deep clone from base data.
 	//
 	// We don't use cloneDeep from lodash here
 	// because we know the data is JSON compatible,
 	// see https://github.com/lodash/lodash/issues/1984
 	const mergedTree = JSON.parse( JSON.stringify( baseData ) );
 
-	const styleKeys = [ 'typography', 'color' ];
+	const styleKeys = [ 'typography', 'color', 'custom', 'spacing' ];
```

Comment:
`custom` and `spacing` are not valid `style` keys, they are only relevant for `settings`. We probably need to port the normalize block schema (and its function) from the ([server](https://github.com/WordPress/gutenberg/blob/master/lib/global-styles.php#L812)).

Reply:
I renamed the variable to validMergeKeys so it can contain keys valid for style and settings. I think ideally we should update the function to no even need to rely on this hardcoded data.

Reply:
But for now, it seems in an acceptable solution following a similar approach to what we had.

Reply:
Hey, I hope you don't mind, but I submitted a commit at https://github.com/WordPress/gutenberg/pull/25795/commits/6652d280f03cae43824148a3ad481a05b369e899 that fixes this (it can be improved in subsequent PRs).

Reply:
Thank you for the enhancements :)